lib/index.js

```diff
@@ -0,0 +1,96 @@
+const config = require('./config');
+
+var request = require('request-promise-native');
+
+
+module.exports = function(apikey) {
+  this.key = apikey;
+
+  this.samlEntity = function(name, entityId, federationId) {
+    this.name = name;
+    this.entityId = entityId;
+    this.federationId = federationId;
+    this.type = "SAML";
+  };
+
+  this.openAthensEntity = function(name, entityId, organizationId, scope) {
+    this.name = name;
+    this.entityId = entityId;
+    this.organizationId = organizationId;
+    this.scope = scope;
+    this.type = "OPEN_ATHENS";
+  };
+
+  this.oAuthEntity = function(provider) {
+    this.provider = provider;
+    // FACEBOOK,
+    // GOOGLE,
+    // TWITTER,
+    // LINKED_IN,
+    // ORCID;
+    this.type = "OAUTH";
+  };
+
+  this.create = (wayfLocal) => {
+    let options = {
+        method: 'POST',
+        uri: config.url+'/device/'+wayfLocal,
+        headers: {
+          'Authorization': 'Token ' + this.key
```

Comment:
I would centralize the header creation. Method that returns the headers or just the Authorization header. This way, name of the header, Authorization, as well as the prefix, Token, is standardized somewhere.